apps/hannk/halide/CMakeLists.txt

```diff
@@ -0,0 +1,156 @@
+
+add_library(common_halide STATIC common_halide.cpp)
+target_link_libraries(common_halide PRIVATE Halide::Halide)
+target_include_directories(common_halide PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>)
+
+set(_HANNK_HALIDE_TARGETS "")
+
+# Function to reduce boilerplate
+function(_add_hannk_halide_library)
+    set(options)
+    set(oneValueArgs TARGET GENERATOR_NAME)
+    set(multiValueArgs SRCS FEATURES GENERATOR_ARGS GENERATOR_DEPS)
+    cmake_parse_arguments(args "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    add_executable(${args_TARGET}.generator ${args_SRCS})
+    target_link_libraries(${args_TARGET}.generator PRIVATE ${args_GENERATOR_DEPS} common_halide Halide::Generator)
+    target_include_directories(${args_TARGET}.generator PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>)
+
+    add_halide_library(${args_TARGET} FROM ${args_TARGET}.generator
+                       NAMESPACE hannk
+                       GENERATOR ${args_GENERATOR_NAME}
+                       FEATURES large_buffers c_plus_plus_name_mangling
+                       PARAMS ${args_GENERATOR_ARGS})
+
+    set(_HANNK_HALIDE_TARGETS ${_HANNK_HALIDE_TARGETS} ${args_TARGET} PARENT_SCOPE)
+endfunction()
+
+_add_hannk_halide_library(
+        TARGET add_uint8_uint8
```

Comment:
Is this losing `no_bounds_query` from the elementwise generators?

Reply:
Those were added after this code is written, so it's less "losing" and more "now we have two build files to maintain"